modules/capi-yaml-infrastructure-rhosp.adoc

```diff
@@ -0,0 +1,107 @@
+<8> Optional: Specify this field but leave it empty to create {rh-openstack} provider-managed security groups with default rules that allow the Kubelet, etcd, and the Kubernetes API server to function correctly. #What are the defaults?#
+Specify additional values to add any required rules to the managed security groups.
+<9> Specify the information associated with the {rh-openstack} security group rule for the cluster.
+You can include the following parameters:
++
+--
+`description`:: Optional: Specify a description of the security group rule.
+`direction`:: Specify the direction of traffic on which to apply the security group rule.
+Valid values for this field are `ingress` or `egress`.
+An `ingress` direction rule applies to incoming traffic and an `egress` direction rule applies to outgoing traffic.
+`etherType`:: Optional: Specify the Internet Protocol type to use.
+Valid values for this field are `IPv6` or `IPv4`.
+`name`:: Specify the name of the security group rule.
+`portRangeMax`:: Optional: Specify a number in the range that is matched by the security group rule.
+If the protocol is TCP or UDP, this value must be less than or equal to the value of the `portRangeMax` attribute.
+`portRangeMin`:: Optional: Specify a number in the range that is matched by the security group rule.
+The `portRangeMin` attribute constrains the `portRangeMax` attribute.
+`protocol`:: Optional: Specify the communication protocol to use.
+For example, to use the Transmission Control Protocol (TCP), specify `tcp`.
+#Need all valid values#
+`remoteGroupID`:: Optional: Specify the remote group ID to associate with this security group rule.
+If you use `remoteGroupID`, do not use `remoteIPPrefix` or `remoteManagedGroups` in the same rule.
+`remoteIPPrefix`:: Optional: Specify the remote IP prefix to associate with this security group rule.
+If you use `remoteIPPrefix`, do not use `remoteGroupID` or `remoteManagedGroups` in the same rule.
+`remoteManagedGroups`:: Optional: Specify the remote managed groups to associate with this security group rule.
+If you use `remoteManagedGroups`, do not use `remoteGroupID` or `remoteIPPrefix` in the same rule.
+--
+<10> Specify the parameters that define your subnet:
++
+--
+`allocationPools`:: Optional: Specify an array of `AllocationPool` objects to apply to the subnet.
+If specified, {rh-openstack} only allocates these IP addresses for machines, but you can create ports from outside of these ranges manually.
+`cidr`:: Specify the IP address range used to create the subnet in Classless Inter-Domain Routing (CIDR) notation, for example `10.0.0.0/24`.
+`dnsNameservers`:: Specify a list of DNS server addresses to use when creating the subnet.
+These addresses must use the same IP version as the `cidr` range.
+--
+
+#Does it make sense to link to upstream optional config options? https://cluster-api-openstack.sigs.k8s.io/clusteropenstack/configuration#optional-configuration#
```

Comment:
In that case, might want to couch it as a warning of some sort: "you might see [these other settings](https://cluster-api-openstack.sigs.k8s.io/clusteropenstack/configuration#optional-configuration) upstream, but we can only attest to the supportability of the ones in this example" or something along those lines

Reply:
Have you considered adding additional validation to prevent downstream users from doing things that we don't support? For example preventing them from using the cloud init parts of the configuration on AWS is something I would like to use ValidatingAdmissionPolicy for